frost-core/src/tests/vectors.rs

```diff
@@ -0,0 +1,274 @@
+
+    let mut signer_nonces: HashMap<Identifier<C>, SigningNonces<C>> = HashMap::new();
+    let mut signer_commitments: HashMap<Identifier<C>, SigningCommitments<C>> = HashMap::new();
+
+    for (i, signer) in round_one_outputs["signers"].as_object().unwrap().iter() {
+        let identifier = u16::from_str(i).unwrap().try_into().unwrap();
+
+        let signing_nonces = SigningNonces::<C> {
+            hiding: Nonce::<C>::from_hex(signer["hiding_nonce"].as_str().unwrap()).unwrap(),
+            binding: Nonce::<C>::from_hex(signer["binding_nonce"].as_str().unwrap()).unwrap(),
+        };
+
+        signer_nonces.insert(identifier, signing_nonces);
+
+        let signing_commitments = SigningCommitments::<C> {
+            identifier,
+            hiding: NonceCommitment::from_hex(signer["hiding_nonce_commitment"].as_str().unwrap())
+                .unwrap(),
+            binding: NonceCommitment::from_hex(
+                signer["binding_nonce_commitment"].as_str().unwrap(),
+            )
+            .unwrap(),
+        };
+
+        signer_commitments.insert(identifier, signing_commitments);
+    }
+
+    // Round two outputs
+
+    let round_two_outputs = &json_vectors["round_two_outputs"];
+
+    let mut signature_shares: HashMap<Identifier<C>, SignatureShare<C>> = HashMap::new();
+
+    for (i, signer) in round_two_outputs["signers"].as_object().unwrap().iter() {
+        let sig_share = <<C::Group as Group>::Field as Field>::Serialization::try_from(
+            hex::decode(signer["sig_share"].as_str().unwrap()).unwrap(),
+        )
+        .unwrap();
+
+        let signature_share = SignatureShare::<C> {
+            identifier: u16::from_str(i).unwrap().try_into().unwrap(),
+            signature: SignatureResponse {
+                z_share: <<C::Group as Group>::Field as Field>::deserialize(&sig_share).unwrap(),
+            },
+        };
+
+        signature_shares.insert(
+            u16::from_str(i).unwrap().try_into().unwrap(),
+            signature_share,
+        );
+    }
+
+    // Final output
+
+    let final_output = &json_vectors["final_output"];
+
+    let signature_bytes = FromHex::from_hex(final_output["sig"].as_str().unwrap()).unwrap();
+
+    (
+        group_public,
+        key_packages,
+        message_bytes,
+        signer_nonces,
+        signer_commitments,
+        group_binding_factor_input,
+        group_binding_factor,
+        signature_shares,
+        signature_bytes,
+    )
+}
+
+/// Test with the given test vectors for a ciphersuite.
+pub fn check_sign_with_test_vectors<C: Ciphersuite + PartialEq>(json_vectors: &Value)
+where
+    <<C::Group as Group>::Serialization as TryFrom<Vec<u8>>>::Error: std::fmt::Debug,
+    <<<C::Group as Group>::Field as Field>::Serialization as TryFrom<Vec<u8>>>::Error:
+        std::fmt::Debug,
+    <<C as Ciphersuite>::SignatureSerialization as std::convert::TryFrom<std::vec::Vec<u8>>>::Error:
+        std::fmt::Debug,
+    C::Group: PartialEq,
```

Comment:
I honestly don't know why the `PartialEq` bounds are required (in fact, I also don't know why the `Debug` bounds are required, but I saw them being used elsewhere). I'll study this later, but it works :stuck_out_tongue:

Reply:
While working on restoring coverage I figured out the `Debug` trait bound issue: `unwrap()` requires that the error type implements `Debug`. We can't currently express that in our `TryFrom` bounds (e.g. `    type Serialization: AsRef<[u8]> + Debug + Default + TryFrom<Vec<u8>>`) because [associated type bounds](https://github.com/rust-lang/rust/issues/52662) are not supported yet. I switched to using the `debugless-unwrap` and that allows us to remove all these Debug bounds.

I'll look into `PartialEq` next :laughing:

Reply:
I figured `PartialEq` out [with SO's help](https://stackoverflow.com/questions/73156316/why-partialeq-bound-is-needed-when-comparing-associated-types). We'll need to manually implement `PartialEq` for some types instead of deriving it. I'll do that in a later PR to make it less confusing.

Reply:
PR here: https://github.com/ZcashFoundation/frost/pull/107